tensorflow/core/kernels/conv_ops_sycl_selectors.h

```diff
@@ -0,0 +1,134 @@
+#ifndef TENSORFLOW_USE_SYCL
+#error This file should only be included when compiling with SYCL support
+#endif
+
+#ifndef TENSORFLOW_KERNELS_CONV_OPS_SYCL_SELECTORS_H_
+#define TENSORFLOW_KERNELS_CONV_OPS_SYCL_SELECTORS_H_
+
+#include "tensorflow/core/kernels/conv_ops_sycl_common.h"
+
+namespace tensorflow {
+class algorithm_selector {
+ public:
+  virtual algorithm get_selection(SYCLConv2DParams const&) = 0;
+};
+template <algorithm Selection>
+class constant_selector final : public algorithm_selector {
+ public:
+  algorithm get_selection(SYCLConv2DParams const&) override {
+    return Selection;
+  }
+};
+using direct_selector = constant_selector<algorithm::direct>;
+using im2col_selector = constant_selector<algorithm::im2col>;
+class winograd_selector final : public algorithm_selector {
+ public:
+  algorithm get_selection(SYCLConv2DParams const& params) override {
+    if (params.stride_rows_ != 1 || params.stride_cols_ != 1) {
+      return algorithm::not_supported;
+    }
+    if (params.window_rows_ == 1 && params.window_cols_ == 3) {
+      return algorithm::winograd_1x3;
+    }
+    if (params.window_rows_ == 3 && params.window_cols_ == 1) {
+      return algorithm::winograd_3x1;
+    }
+    if (params.window_rows_ == 3 && params.window_cols_ == 3) {
+      return algorithm::winograd_3x3;
+    }
+    return algorithm::not_supported;
+  }
+};
+class direct_tiled_selector final : public algorithm_selector {
+ public:
+  algorithm get_selection(SYCLConv2DParams const& params) override {
+    if (params.window_rows_ != params.window_cols_ ||
+        params.stride_rows_ != params.stride_cols_) {
+      return algorithm::not_supported;
+    }
+    if (params.window_rows_ == 1 && params.stride_rows_ == 2) {
```

Comment:
Or statements instead of having 6 if. It shouild matter for the result but it makes it clearer that it's either going to call direct_tiled or not_supported.

Reply:
I think this was meant to be changed in the future. Again I don't think we really care about cleaning the conv_ops code right now unless there is a major issue.

Reply:
we will be replacing Conv code rather soon. So I am not too bothered with spending too much time on it.